scripts/xapi-domains.service

```diff
@@ -1,9 +1,9 @@
 [Unit]
 Description=Start/stop domains on dom0 start/shutdown
 Requires=proc-xen.mount
-Wants=xapi-init-complete.target
+Wants=xapi-init-complete.target stunnel@xapi.service
```

Comment:
i've been testing with only 'After += stunnel@xapi.service', but i think we probably also want this ^

(`Wants` rather than `Requires`, since we if we are starting VMs on the master we shouldn't need stunnel)

Reply:
"Wants" is probably fine. We don't want a "Requires" because that'd also stop xapi-domains, if stunnel is stopped which would be bad because we do have code that restarts stunnel and we wouldn't expect VMs to be shutdown as a side-effect.

Reply:
I think it would be good to have this on its own line - it makes patches easier

Reply:
> "Wants" is probably fine. We don't want a "Requires" because that'd also stop xapi-domains, if stunnel is stopped which would be bad because we do have code that restarts stunnel and we wouldn't expect VMs to be shutdown as a side-effect.

i'm struggling to find documentation that says systemd will automatically stops its services. can you convince me that adding a `Requires=stunnel@xapi.service` would cause that behaviour?

my understanding of the difference between `Requires` and `Wants` is that `Requires` adds a hard dependency, whereas `Wants` adds a soft one

Reply:
What soft and hard dependencies mean:
```
Wants=
Units listed in this option will be started if the configuring unit is. However, if the listed units fail to start or cannot be added to the transaction, this has no impact on the validity of the transaction as a whole, and this unit will still be started.
```
```
Requires=
If this unit gets activated, the units listed will be activated as well. If one of the other units fails to activate, and an ordering dependency After= on the failing unit is set, this unit will not be started. Besides, with or without specifying After=, this unit will be stopped if one of the other units is explicitly stopped.
```

Reply:
i think only `After/Before` will affect the shutdown order

Reply:
That's what I understood from the documentation as well.